src/reviews/useCases/getReviewByUserId/GetReviewByUserIdUseCase.ts

```diff
@@ -0,0 +1,24 @@
+import { GetReviewByUserIdResponse } from './GetReviewByUserIdResponse';
+import { GetReviewByUserIdRepositoryInterface } from '../../repos/getReviewByUserId/GetReviewByUserIdRepositoryInterface';
+
+export class GetReviewByUserIdUseCase {
+  constructor(
+    private readonly reviewRepository: GetReviewByUserIdRepositoryInterface
+  ) {}
+
+  async execute(userId: string): Promise<GetReviewByUserIdResponse> {
+    const retrieveReview = await this.reviewRepository.getReviewByUserId(
+      userId
+    );
+    try {
+      if (retrieveReview == undefined) throw new Error('Review not found');
+      return {
+        id: retrieveReview.getId(),
+        userId: retrieveReview?.getUserId(),
+        userReview: retrieveReview?.getUserReview(),
+      };
+    } catch (error) {
+      throw new Error((error as Error).message);
+    }
+  }
```

Comment:
I think we can remove try/ catch here so that the code maintains readability and simplifies the error handling.